api/handlers_params.go

```diff
@@ -0,0 +1,379 @@
+		return
+	}
+
+	prefix := vars["prefix"]
+	if prefix == "" {
+		handleError(w, apierror.New(apierror.ErrNotFound, "prefix is required", nil))
+		return
+	}
+
+	path := fmt.Sprintf("/%s/%s", s.org, prefix)
+	params, err := ssmService.ListParametersByPath(r.Context(), path)
+	if err != nil {
+		msg := fmt.Sprintf("unable to list params from the ssm service prefix %s", prefix)
+		handleError(w, errors.Wrap(err, msg))
+		return
+	}
+
+	for _, p := range params {
+		err := ssmService.DeleteParameter(r.Context(), p)
+		if err != nil {
+			msg := fmt.Sprintf("unable to delete all parameters from the ssm service path %s, failed to delete %s", path, p)
+			handleError(w, errors.Wrap(err, msg))
+			return
+		}
+	}
+
+	j, err := json.Marshal(
+		struct {
+			Message string
+			Deleted int
+		}{
+			Message: "OK",
+			Deleted: len(params),
+		})
+	if err != nil {
+		handleError(w, errors.Wrap(err, "unable to marshal response from the ssm service"))
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+	w.Write(j)
+}
+
+// ParamUpdateHandler updates a parameter store parameter
+func (s *server) ParamUpdateHandler(w http.ResponseWriter, r *http.Request) {
+	w = LogWriter{w}
+	vars := mux.Vars(r)
+	account := vars["account"]
+	ssmService, ok := s.ssmServices[account]
+	if !ok {
+		msg := fmt.Sprintf("ssm service not found for account: %s", account)
+		handleError(w, apierror.New(apierror.ErrNotFound, msg, nil))
+		return
+	}
+
+	prefix := vars["prefix"]
+	if prefix == "" {
+		handleError(w, apierror.New(apierror.ErrNotFound, "prefix is required", nil))
+		return
+	}
+
+	paramName := vars["param"]
+	if paramName == "" {
+		handleError(w, apierror.New(apierror.ErrNotFound, "param is required", nil))
+		return
+	}
+
+	input := &ssm.PutParameterInput{}
+	err := json.NewDecoder(r.Body).Decode(&input)
+	if err != nil {
+		msg := fmt.Sprintf("cannot decode body into put parameter input: %s", err)
+		handleError(w, apierror.New(apierror.ErrBadRequest, msg, err))
+		return
+	}
+	defer r.Body.Close()
+
+	path := fmt.Sprintf("/%s/%s/%s", s.org, prefix, paramName)
+	parameter, err := ssmService.GetParameter(r.Context(), path)
+	if err != nil {
+		msg := fmt.Sprintf("unable to get parameter from the ssm service path %s", path)
+		handleError(w, errors.Wrap(err, msg))
+		return
+	}
+
+	tags, err := ssmService.ListParameterTags(r.Context(), aws.StringValue(parameter.Name))
+	if err != nil {
+		msg := fmt.Sprintf("unable to get tags from the ssm service for param %s", path)
+		handleError(w, errors.Wrap(err, msg))
+		return
+	}
+
+	// check the org tag for the parameter
+	org := ""
+	for _, t := range tags {
+		if aws.StringValue(t.Key) == "spinup:org" || aws.StringValue(t.Key) == "yale:org" {
+			log.Debugf("found org tag: %s:%s", aws.StringValue(t.Key), aws.StringValue(t.Value))
+			org = aws.StringValue(t.Value)
+		}
+	}
+	if org != s.org {
+		handleError(w, apierror.New(apierror.ErrForbidden, "wrong org", nil))
+		return
+	}
```

Comment:
I didn't see this check in the other handlers, shouldn't it be everywhere?

Reply:
Possibly, it's going to be a lot of calls to the API for checking the org when doing a list...  Since the org is forced to be part of the param name in each handler, I *think* its actually ok to remove this check here and not use it.